### ISSUE TITLE
DDL job always run.I can't kill it!

### ISSUE THREAD

**Issue by 7le** [pingcap/tidb#4986]
Please answer these questions before submitting your issue. Thanks!

1. What did you do?
If possible, provide a recipe for reproducing the error.
Look at Id 46.I want to add a index,but the ddl job always run,I can't kill it.
```
tidb> admin show ddl jobs;
+------------------------------------------------------------------------------------------------------------------------------+---------+
| JOBS                                                                                                                         | STATE   |
+------------------------------------------------------------------------------------------------------------------------------+---------+
| ID:46, Type:add index, State:running, SchemaState:write reorganization, SchemaID:1, TableID:33, RowCount:165059980, ArgLen:0 | running |
| ID:47, Type:drop index, State:none, SchemaState:none, SchemaID:1, TableID:33, RowCount:0, ArgLen:0                           | none    |
| ID:48, Type:drop index, State:none, SchemaState:none, SchemaID:1, TableID:33, RowCount:0, ArgLen:0                           | none    |
| ID:45, Type:drop column, State:synced, SchemaState:none, SchemaID:1, TableID:33, RowCount:0, ArgLen:0                        | synced  |
| ID:44, Type:drop column, State:synced, SchemaState:none, SchemaID:1, TableID:33, RowCount:0, ArgLen:0                        | synced  |
| ID:43, Type:drop column, State:synced, SchemaState:none, SchemaID:1, TableID:33, RowCount:0, ArgLen:0                        | synced  |
| ID:42, Type:add column, State:synced, SchemaState:public, SchemaID:1, TableID:33, RowCount:0, ArgLen:0                       | synced  |
| ID:41, Type:add column, State:synced, SchemaState:public, SchemaID:1, TableID:33, RowCount:0, ArgLen:0                       | synced  |
| ID:40, Type:drop index, State:synced, SchemaState:none, SchemaID:1, TableID:33, RowCount:0, ArgLen:0                         | synced  |
| ID:39, Type:add column, State:synced, SchemaState:public, SchemaID:1, TableID:33, RowCount:0, ArgLen:0                       | synced  |
| ID:38, Type:add index, State:synced, SchemaState:public, SchemaID:1, TableID:33, RowCount:100283953, ArgLen:0                | synced  |
| ID:37, Type:drop table, State:synced, SchemaState:none, SchemaID:1, TableID:35, RowCount:0, ArgLen:0                         | synced  |
| ID:36, Type:create table, State:synced, SchemaState:public, SchemaID:1, TableID:35, RowCount:0, ArgLen:0                     | synced  |
+------------------------------------------------------------------------------------------------------------------------------+---------+
```

2. What did you expect to see?
I want the id 46 job success and stop.

3. What did you see instead?
The total number of my table is 100283953 , but the Id 46 job  show ``RowCount:165059980.``  and the number still increase .

I want to kill it. But i can't find the session Id.
```
tidb> show processlist;
+------+------+---------------+------+---------+------+-------+------------------+
| Id   | User | Host          | db   | Command | Time | State | Info             |
+------+------+---------------+------+---------+------+-------+------------------+
|    2 | root | 127.0.0.1     | test | Query   |    0 | 2     | show processlist |
|    4 | root | 192.168.10.72 | test | Query   |    0 | 2     |                  |
+------+------+---------------+------+---------+------+-------+------------------+
2 rows in set (0.00 sec)

```

4. What version of TiDB are you using (`tidb-server -V`)?
```
tidb> select tidb_version();
+----------------------------------------------------------------------------------------------------------------------------------------------------------+
| tidb_version()                                                                                                                                           |
+----------------------------------------------------------------------------------------------------------------------------------------------------------+
| Release Version: v1.0.1-1-gaee0b39
Git Commit Hash: aee0b39af29c87742811f02c657dcf2b0fe1d60c
Git Branch: release-1.0
UTC Build Time: 2017-11-01 01:51:09 |
+----------------------------------------------------------------------------------------------------------------------------------------------------------+
1 row in set (0.00 sec)

```

**Comment by shenli** [comment]
@7le Could you send me your tidb-server's log and the schema/ddl statement to shenli@pingcap.com?

**Comment by coocood** [comment]
There is a cancel statement:
```
ADMIN CANCEL DDL JOBS 'job_id';
``` 

In this case, you can try:
```
ADMIN CANCEL DDL JOBS '46';
```

**Comment by coocood** [comment]
And kill the session doesn't stop the DDL worker.

**Comment by 7le** [comment]
@coocood  Thanks.  I kill it successfully.

**Comment by 7le** [comment]
@shenli  Thanks. The question is been resolved. So I don't send logs.

**Comment by 812002343** [comment]
OMG

**Comment by shenli** [comment]
@7le May I have the `show create table your_table_name;` of the table which you run AddIndex operation?

**Comment by 7le** [comment]
@shenli Of course. It's in the company's server. I will send you the table and the tidb-server's log tomorrow.

**Comment by shenli** [comment]
@7le Thanks a lot!

**Comment by shenli** [comment]
@7le The `RowCount` in `Admin show ddl` is not accurate in the previous version. This is fixed in #5041 .  We will cherry-pick this patch to release-1.0 .

**Comment by 7le** [comment]
@shenli  I got it . Thanks.  😃😃
